### PR TITLE
add API stubbing for addAsset and test gh action

### LINKED ISSUE #1653: [SPIKE] Investigate splitting e2e tests into stubbed vs full e2e mode
Currently in our e2e tests, we use a Testnet account to run through a few scenarios like sending a payment, adding a trustline, etc. These scenarios generate tx's and actually submit to Testnet.

The benefits of re-using the same account and not stubbing out these calls are:
1. We get signal that all our integrated API's are continuing to work as we expect. This gives us a more complete picture of if Freighter is actually working for the end user and we're not missing anything with some stubbing that is overly optimistic 
2. Tests run a bit faster because we don't need to be constantly creating and funding new accounts to test with
3. For Soroban token testing, we don't need to worry about minting to this newly generated account
4. Having to stub out calls makes creating tests slower and more complex. It's very fast to create an e2e test without stubbing. This is important as we really need a lot more coverage of our code.

The cons are:
1. The tests are non-deterministic. If a test fails in an unexpected way, the test account might end up in a weird state that causes subsequent test failures. This could be come even more of a problem with multiple devs running tests at the same time
2. Issues with any 3rd party API, even if transient, can cause a test failure. For example, in times of high congestion on Testnet, a test could fail just for taking too long

Possible solution:
We default to stubbing out these calls when the tests run in CI and, maybe before we cut a beta/prod release, we flip a toggle that allows these tests to run against actual API's. These full API test runs would happen infrequently and by only one dev at a time.

Open Questions:
1. How easy/hard is it to create this kind of toggle in Playwright? 
2. Can we run it from a Github Action?
3. How much work/effort will it be to stub out all the existing calls in our current e2e tests?

### PR DESCRIPTION
**Pull request by piyalbasu**
Closes #1653 

This is a strategy for being able to run our e2e tests in 2 different modes:

1) Run our tests making calls to services like Horizon, stellar.expert, etc. This is useful for making sure our integrations are actually working as expected and our tests are truly mimicking a real user's experience

2) Run our tests stubbing out the calls to external services. This is useful for making sure that we don't get false positives from those services having issues, experiencing network latency, etc. When we run our tests over and over again, we can sometimes hit these type of issues which slow down our development time as we need to investigate whether there's a bug or if it's just the test being flakey. Also, because we're making changes to a real account on Testnet, we can end up in an unexpected state if a test fails or if multiple people are running the same test at once.

This approach will run our tests in Mode 2 (stubbed mode) when we're working on PR's to a `release` branch. We anticipate these tests being run frequently as people open PR's/add commits, so we want these tests to be as stable as possible.

For PR's to `master` (.i.e., a change that we're planning on sending to production soon), we will run these tests in Mode 1, where we can get an actual sense of how the app is working using real services. We anticipate that this version of the tests will run more infrequently - only when a PR has been fully reviewed and approved, and pushed to the release branch.

Let me know your thoughts on this approach and if I'm missing anything here!

### CODE REVIEW

File: extension/e2e-tests/addAsset.test.ts [hunk 1]
```diff
@@ -34,13 +64,12 @@ test.skip("Adding unverified Soroban token", async ({ page, extensionId }) => {
...
 });
 test("Adding Soroban verified token", async ({ page, extensionId }) => {
-  const assetsList = await fetch(
-    "https://api.stellar.expert/explorer/testnet/asset-list/top50",
-  );
-  const assetsListData = await assetsList.json();
-  const verifiedToken =
-    assetsListData?.assets[0]?.contract ||
-    "CBIELTK6YBZJU5UP2WWQEUCYKLPU6AUNZ2BQ4WWFEIE3USCIHMXQDAMA";
+  if (!process.env.IS_INTEGRATION_MODE) {
```
Comment (piyalbasu) [inline extension/e2e-tests/addAsset.test.ts:67]: we key off this param to decide whether to stub or not

File: .github/workflows/runTests.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - run: yarn build:freighter-api
       - run: yarn build:extension
       - run: yarn test:ci
-      - run: yarn test:e2e
+      - run: ${{ github.base_ref == 'master' && 'IS_INTEGRATION_MODE=true' || '' }} yarn test:e2e
```
Comment (piyalbasu) [inline .github/workflows/runTests.yml:23]: for PR's based on `master`, we'll run the test in integration mode

File: extension/e2e-tests/addAsset.test.ts [hunk 3]
```diff
@@ -34,13 +64,12 @@ test.skip("Adding unverified Soroban token", async ({ page, extensionId }) => {
   await expect(page.getByTestId("account-view")).toContainText("E2E");
 });
 test("Adding Soroban verified token", async ({ page, extensionId }) => {
-  const assetsList = await fetch(
-    "https://api.stellar.expert/explorer/testnet/asset-list/top50",
-  );
-  const assetsListData = await assetsList.json();
-  const verifiedToken =
-    assetsListData?.assets[0]?.contract ||
-    "CBIELTK6YBZJU5UP2WWQEUCYKLPU6AUNZ2BQ4WWFEIE3USCIHMXQDAMA";
```
Comment (piyalbasu) [inline extension/e2e-tests/addAsset.test.ts:43]: I'm removing some unnecessary data fetching here - it's more reliable to just hardcode the USDC token address

**Comment by aristidesstaffieri** [comment]
I think generally this lgtm. One thought I have about this is that it could be nice to try to get all of the the things that rely on `IS_INTEGRATION_MODE` to toggle modes in one place.
We could import all of those deps into a common test file that uses the toggle and then import the toggled deps into the tests in order to keep from littering this logic all through the tests.

**Comment by piyalbasu** [comment]
> I think generally this lgtm. One thought I have about this is that it could be nice to try to get all of the the things that rely on `IS_INTEGRATION_MODE` to toggle modes in one place. We could import all of those deps into a common test file that uses the toggle and then import the toggled deps into the tests in order to keep from littering this logic all through the tests.

Oh this is an interesting thought - I added a fixture in this commit to centralize all of our stubbing logic. There will probably still be instances where we need to override these base stubs to simulate certain interactions, but I think this might work for simple on/off situations

https://github.com/stellar/freighter/pull/1749/commits/dd34a60f068f0dd6bf233ddb6f41c251341ad9fc

**Comment by aristidesstaffieri** [comment]
> > I think generally this lgtm. One thought I have about this is that it could be nice to try to get all of the the things that rely on `IS_INTEGRATION_MODE` to toggle modes in one place. We could import all of those deps into a common test file that uses the toggle and then import the toggled deps into the tests in order to keep from littering this logic all through the tests.
> 
> Oh this is an interesting thought - I added a fixture in this commit to centralize all of our stubbing logic. There will probably still be instances where we need to override these base stubs to simulate certain interactions, but I think this might work for simple on/off situations
> 
> [dd34a60](https://github.com/stellar/freighter/commit/dd34a60f068f0dd6bf233ddb6f41c251341ad9fc)

nice thanks yeah I recognize that it'll be hard to make this a 100% rule but it should help the readability a bit imo.

**Review by aristidesstaffieri — APPROVED** [review]

**Pull request merged**